crates/uv-auth/src/credentials.rs

```diff
@@ -1,117 +1,156 @@
 use base64::prelude::BASE64_STANDARD;
+use base64::read::DecoderReader;
 use base64::write::EncoderWriter;
-use netrc::Authenticator;
+
+use netrc::Netrc;
 use reqwest::header::HeaderValue;
+use reqwest::Request;
+use std::io::Read;
 use std::io::Write;
 use url::Url;
 
 #[derive(Clone, Debug, PartialEq)]
 pub(crate) struct Credentials {
-    username: String,
+    username: Option<String>,
     password: Option<String>,
```

Comment:
You could probably make this `Option<Box<str>>` if you were excited about shrinking the struct size :joy:

Reply:
I made it `Option` for clarity because it's passed around as an empty string (`""`) a lot in reqwest and it simplified the logic to encode that as `None`.

Reply:
Yeah `Option` is cool, this makes sense. (Maybe document that it's guaranteed to be non-empty?) I was commenting on `String` vs. `Box<str>` but it doesn't really matter.